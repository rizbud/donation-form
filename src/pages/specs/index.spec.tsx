import { screen, render } from "@testing-library/react";

import Index from "..";

describe("Index", () => {
  it("renders successfully", () => {
    expect(() => {
      render(<Index />);
    }).not.toThrow();
  });

  it("renders a form", () => {
    render(<Index />);

    expect(screen.getByText("Amount")).toBeInTheDocument();
    expect(screen.getByText("Name")).toBeInTheDocument();
    expect(screen.getByText("Email")).toBeInTheDocument();
    expect(screen.getByText("ID Number (NRIC/FIN)")).toBeInTheDocument();
    expect(screen.getByText("Postal Code")).toBeInTheDocument();
    expect(screen.getByText("Unit Number")).toBeInTheDocument();
    expect(screen.getByText("Address")).toBeInTheDocument();
    expect(screen.getByText("Remarks")).toBeInTheDocument();
    expect(screen.getByRole("button")).toBeInTheDocument();
  });
});

// Snapshot test
describe("Index", () => {
  it("renders successfully", () => {
    const { baseElement } = render(<Index />);

    expect(baseElement).toMatchSnapshot();
  });
});
