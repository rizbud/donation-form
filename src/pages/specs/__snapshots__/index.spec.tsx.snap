// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders successfully 1`] = `
<body>
  <div>
    <main
      class="font-sans min-h-screen bg-gray-100 overflow-x-hidden lg:overflow-x-auto"
    >
      <div
        class="w-full xl:max-w-screen-lg 2xl:max-w-screen-xl mx-auto my-5"
      >
        <div
          class="max-w-xl mx-auto py-4"
        >
          <div
            class="px-6 rounded-lg gap-6"
          >
            <form>
              <div
                class="bg-white shadow-md pt-6 pb-1 px-6 rounded-lg mb-6 relative"
              >
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-1"
                  >
                    Amount
                    <span
                      class="text-red-500 text-xs font-normal"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="relative flex items-center"
                  >
                    <span
                      class="absolute inset-y-0 left-0 flex items-center pb-px"
                    >
                      $
                    </span>
                    <input
                      aria-invalid="false"
                      aria-label="Amount"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary pl-4"
                      placeholder="10"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-1"
                  >
                    Name
                    <span
                      class="text-red-500 text-xs font-normal"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="relative flex items-center"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Name"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      name="name"
                      placeholder="John Doe"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-1"
                  >
                    Email
                    <span
                      class="text-red-500 text-xs font-normal"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="relative flex items-center"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Email"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      name="email"
                      placeholder="johndoe@gmail.com"
                      type="email"
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-1"
                  >
                    ID Number (NRIC/FIN)
                    <span
                      class="text-red-500 text-xs font-normal"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="relative flex items-center"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="ID Number (NRIC/FIN)"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      maxlength="9"
                      name="id_number"
                      placeholder="S1234567A"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-1"
                  >
                    Postal Code
                    <span
                      class="text-red-500 text-xs font-normal"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="relative flex items-center"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Postal Code"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      maxlength="6"
                      name="postal_code"
                      placeholder="123456"
                      type="number"
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-1"
                  >
                    Unit Number
                    <span
                      class="text-red-500 text-xs font-normal"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="relative flex items-center"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Unit Number"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      name="unit_number"
                      placeholder="123-45"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-2"
                  >
                    Address
                  </label>
                  <div
                    class="relative"
                  >
                    <textarea
                      aria-invalid="false"
                      aria-label="Address"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      name="address"
                      placeholder="Street Name
Block Number, Building Name"
                      rows="2"
                    />
                  </div>
                </div>
                <div
                  class="mb-6"
                >
                  <label
                    class="block text-sm font-semibold text-slate-800 mb-2"
                  >
                    Remarks
                  </label>
                  <div
                    class="relative"
                  >
                    <textarea
                      aria-invalid="false"
                      aria-label="Remarks"
                      class="placeholder:text-gray-400 text-black border-gray-300 block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
                      name="remarks"
                      placeholder="Optional"
                      rows="1"
                    />
                  </div>
                </div>
              </div>
              <div>
                <button
                  class="bg-primary text-white font-regular rounded-lg px-5 py-3 hover:opacity-80 disabled:opacity-50 w-full"
                  type="submit"
                >
                  SEND
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>
</body>
`;
